components/feature/autofill/src/main/java/mozilla/components/feature/autofill/response/fill/LoginFillResponseBuilder.kt

```diff
@@ -55,6 +56,26 @@ internal data class LoginFillResponseBuilder(
             searchDatasetBuilder.build(context, configuration, imeSpec)
         )
 
+        builder.handleSaveInfo(parsedStructure)
+
         return builder.build()
     }
 }
+
+@RequiresApi(Build.VERSION_CODES.O)
+internal fun FillResponse.Builder.handleSaveInfo(ps: ParsedStructure) {
+    val ids = arrayOf(ps.usernameId, ps.passwordId).filterNotNull()
+    if (ids.isNotEmpty()) {
+        setSaveInfo(
+            SaveInfo.Builder(
+                SaveInfo.SAVE_DATA_TYPE_USERNAME or SaveInfo.SAVE_DATA_TYPE_PASSWORD,
+                ids.toTypedArray()
+            ).also {
+                if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.Q && ps.passwordId == null) {
+                    // Continue workflow until password field is reached
+                    it.setFlags(SaveInfo.FLAG_DELAY_SAVE)
```

Comment:
Can you help me understand how the flag works? Will Android collect all values until we do not "delay" anymore? 🤔

Reply:
Correct.
If we set the flag then save UI is not triggered and `request.fillContexts` is delivered in future requests with `AutofillService.onFillRequest()` and `AutofillService.onSaveRequest()`
It's [here](https://developer.android.com/reference/android/service/autofill/SaveInfo#FLAG_DELAY_SAVE)